draft-ietf-opsawg-pcapng.md

```diff
@@ -1463,7 +1463,10 @@ following:
 | 0-1 | Inbound / Outbound packet (00 = information not available, 01 = inbound, 10 = outbound) |
 | 2-4 | Reception type (000 = not specified, 001 = unicast, 010 = multicast, 011 = broadcast, 100 = promiscuous). |
 | 5-8 | FCS length, in octets (0000 if this information is not available). This value overrides the if_fcslen option of the Interface Description Block, and is used with those link layers (e.g. PPP) where the length of the FCS can change during time. |
-| 9-15 | Reserved (MUST be set to zero). |
+| 9 | Csum NotReady, as a consequence the checksum field in the transport header is not valid, and checksum validation on the current packet should not be performed. |
```

Comment:
"as a consequence" of what?

As a consequence of checksum offloading, e.g. a transmitted packet on an interface doing checksum offloading, so that the host networking stack doesn't compute and fill in the checksum before handing the packet either to the network adapter *or* the wraparound code path in the packet capture mechanism?

Or also as a consequence of segmentation/desegmentation offloading (or quasi-offloading, as I have the impression may be done in the Linux networking for adapters that *don't* do segmentation/desegmentation offloading)?

Reply:
"As a consequence of checksum or segmentation offloading" will be more accurate, as it also sets CHECKSUM_PARTIAL (Csum NotReady) for the packets to segmentation offloading and does checksum in network adapter later.

Reply:
> and does checksum in network adapter later.

That's called "checksum offloading".  If you offload segmentation, so that the TCP segment you hand to the adapter is different from the TCP segments, possibly plural, that go out on the network, the TCP checksum has to be computed after the segmentation.

And, when multiple TCP segments are received by the adapter and reassembled, with the reassembled segment being what's handed to the host, the adapter has little reason to recompute a checksum for the packet it hands to the host.

(Well, if you don't trust the I/O bus, packet checksums can come in handy; in the early days of NetApp, they sent out UDPgrams without a checksum, but it turned out that the UDPgram was getting mangled when read by the adapter over EISA, so they switched to providing a checksum.)

Reply:
reposted with the suggested words.